### PR TITLE
Docs (Guides): Mat elevation guide

### LINKED ISSUE #2800: Add elevation guide
#### Bug, feature request, or proposal:
proposal

#### What is the expected behavior?
That `.md-elevation-zX` classes are documented.

#### What is the current behavior?
No documentation exists other than within closed [PR](https://github.com/angular/material2/pull/2212) and [issue](https://github.com/angular/material2/pull/222/files) comments. There are some [design documents](https://docs.google.com/document/d/1W3NGSLqDZzjbBBLW2C6y_6NUxtvdZAVaJvg58LY3Q0E/edit?usp=sharing), but they aren't really designed for developers using the APIs.

#### What is the use-case or motivation for changing an existing behavior?
To avoid repeated [questions](https://github.com/angular/material2/pull/222#issuecomment-243015295) and rejected PRs like https://github.com/angular/material2/pull/2212.

#### Is there anything else we should know?
I would like to help with this documentation, but I would like some feedback on how the team would like this to be done before I submit a PR.

Should there be a separate doc/guide for elevation that covers examples of multiple components?

Should there be examples of using elevation in each component that it applies to?
Is there a list of such components?

### PR DESCRIPTION
**Pull request by M-a-c**
Added a basic guide on how to use Mat Elevation.
Includes Example
* Link to Material designs spec for Elevation
* How to use the CSS class
* How to use the mixin
* Deprecated naming

Fixes #2800

### CODE REVIEW

File: guides/mat-elevation.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+Elevation is implemented with a class, simply adding the class `mat-elevation-z#` where # is the elevation number you want, 0-24.
+
+## Example
+<!-- example(elevation-overview) -->
+
+
+## Mixins
+In order to use the mixin for Elevation you must 
+`@import '~@angular/material/core/style/_elevation.scss';`
```
Comment (willshowell) [inline guides/mat-elevation.md]: I think this should be `~@angular/material/theming` now that they're using a flat release structure (#3776)

Comment (M-a-c) [inline guides/mat-elevation.md]: Ok yeah I'll have to fix this then

Comment (M-a-c) [inline guides/mat-elevation.md]: @willshowell Do you know if i will need to specify elevation.scss from that path?

Comment (willshowell) [inline guides/mat-elevation.md]: @M-a-c no that should error. Using the master builds, check out your node_modules. At `/node_modules/@angular/material/_theming.scss` you can see that all the mixins/functions/definitions are concatenated together, including `mat-elevation()`.

And due to [Sass importing of partials](http://sass-lang.com/guide#topic-4), you don't need the leading underscore or file extension in your import statement.

File: guides/mat-elevation.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+In order to use the mixin for Elevation you must 
+`@import '~@angular/material/theming';`
+$zValue must be a value between 0 and 24, inclusive.
+`@include mat-elevation($zValue);`
+
+
+How to use the mixin 
+```scss
+.myClass {
+  @include $mat-elevation(2);
```
Comment (Sabartius) [inline guides/mat-elevation.md]: shouldn't this be without the `$`? -> `@include mat-elevation(2);`

File: guides/mat-elevation.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  &:active {
+    @include $mat-elevation(8);
```
Comment (Sabartius) [inline guides/mat-elevation.md]: same here

File: src/examples/elevation-overview/elevation-overview-example.scss [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    text-align: center;
+    height: 50px;
+    margin-left: auto;
+    margin-right: auto;
+    background-color: lightblue;
+    margin-top: 10px;
+}
+
+.dynamimc {
+    @include $mat-elevation(2);
```
Comment (Sabartius) [inline src/examples/elevation-overview/elevation-overview-example.scss]: Also without `$` ? At least in my Project it won't compile with a `$`

**Comment by M-a-c** [comment]
@Sabartius thanks for the feedback

File: src/examples/elevation-overview/elevation-overview-example.html [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<div class="dynamimc">Hover over me!</div>
```
Comment (Splaktar) [inline src/examples/elevation-overview/elevation-overview-example.html]: Did you mean dynamic?

File: src/examples/elevation-overview/elevation-overview-example.scss [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    text-align: center;
+    height: 50px;
+    margin-left: auto;
+    margin-right: auto;
+    background-color: lightblue;
+    margin-top: 10px;
+}
+
+
+.dynamimc {
```
Comment (Splaktar) [inline src/examples/elevation-overview/elevation-overview-example.scss]: Ditto

File: src/examples/elevation-overview/elevation-overview-example.scss [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    @include mat-elevation(2);
+
+    &:active {
```
Comment (willshowell) [inline src/examples/elevation-overview/elevation-overview-example.scss]: Shouldn't this be `&:hover`?

File: guides/mat-elevation.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  @include mat-elevation(2);
+
+  &:active {
+    @include mat-elevation(8);
+  }
+}
+```
+
+## Deprecated
+The use of `mdElevation($zValue)` is deprecated.
```
Comment (Splaktar) [inline guides/mat-elevation.md]: Should we use the words attribute / directive in this text to be more clear?

File: src/examples/elevation-overview/elevation-overview-example.html [hunk 9]
```diff
@@ -0,0 +1,5 @@
...
+
+<div class"no-elevation">No Elevation</div>
```
Comment (Splaktar) [inline src/examples/elevation-overview/elevation-overview-example.html]: Does adding the `no-elevation` class here create confusion?

Comment (M-a-c) [inline src/examples/elevation-overview/elevation-overview-example.html]: Yeah good point, I removed it in the next commit

**Comment by Splaktar** [comment]
@ladyleet any additional comments/suggestions for this PR?

File: src/examples/elevation-overview/elevation-overview-example.scss [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+
+.dynamic {
+    @include mat-elevation(2);
+
+    &:hover {
+        @include mat-elevation(8);
+    }
+}
```
Comment (mmalerba) [inline src/examples/elevation-overview/elevation-overview-example.scss]: newline

File: guides/mat-elevation.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
+# mat-elevation
```
Comment (mmalerba) [inline guides/mat-elevation.md]: how do you feel about this?
````markdown
# Elevation
Angular Material's elevation classes and mixins allow you to add separation between elements along the z-axis.
All material design elements have resting elevations. In addition, some elements may change their elevation in response to user interaction. The [Material Design spec](https://material.io/guidelines/material-design/elevation-shadows.html) explains how best to use elevation.

Angular Material provides two ways to control the elevation of elements: predefined CSS classes and mixins.

## Predefined CSS classes
The easiest way to add elevation to an element is to simply add one of the predefined CSS classes `mat-elevation-z#` where `#` is the elevation number you want, 0-24. Dynamic elevation can be acheived by switching elevation classes:

```html
<div [class.mat-elevation-z2]="!isActive" [class.mat-elevation-z8]="isActive"></div>
```

<!-- example(elevation-overview) -->

## Mixins
Elevations can also be added in CSS via the `mat-elevation` mixin, which takes a number 0-24 indicating the elevation level of the element. In order to use the mixin you must import `~@angular/material/theming`.

```scss
@import ~@angular/material/theming;

.myClass {
  @include mat-elevation(2);

  &:active {
    @include mat-elevation(8);
  }
}
```
````

**Comment by mmalerba** [comment]
Closing due to inactivity

**Pull request closed without merging**